src/main/java/com/epam/ta/reportportal/core/integration/impl/GetIntegrationHandlerImpl.java

```diff
@@ -56,4 +67,30 @@ public IntegrationResource getGlobalIntegrationById(Long integrationId) {
 
 		return IntegrationConverter.TO_INTEGRATION_RESOURCE.apply(integration);
 	}
+
+	@Override
+	public Optional<Integration> findEnabledByProjectIdOrGlobalAndIntegrationGroup(Long projectId, IntegrationGroupEnum integrationGroup) {
+
+		List<Long> integrationTypeIds = integrationTypeRepository.findAllByIntegrationGroup(integrationGroup)
+				.stream()
+				.map(IntegrationType::getId)
+				.collect(Collectors.toList());
+
+		List<Integration> integrations = integrationRepository.findAllByProjectIdAndInIntegrationTypeIds(projectId, integrationTypeIds);
+
+		if (!CollectionUtils.isEmpty(integrations)) {
+
+			return integrations.stream()
+					.filter(Integration::isEnabled)
+					.findFirst();
+
+		} else {
+
+			return integrationRepository.findAllGlobalInIntegrationTypeIds(integrationTypeIds)
```

Comment:
what about using private methods for better reading?

Reply:
done